### ISSUE TITLE
Grant access to specific users or groups?

### ISSUE THREAD

**Issue by datkinson-mdvip** [microsoft/azurechat#326]
Is it possible to grant access to specific users or groups within the current design? Is it as simple as checking "Assignment Required" in the Enterprise Application and adding the user or group assignment?

**Comment by bwitzig-zen** [comment]
This depends on how you are managing auth. Assuming you are using 365, you change the settings in the azure enterprise app section for this app to "require assignment" and then add the users/groups.

**Comment by olafwrieden** [comment]
Correct! Some teams tend to use the Enterprise Application as the point of permission assignment (e.g. everyone in a specific Security Group). This applies when you configure Entra ID authentication. Alternatively, you could lock it down at the App Service level using a similar approach, or handle access control in the application itself where you could intercept the NextAuth / Auth.js sign in/up callback. Then you could perform a check on the email address of the user and, for example, check if they match the `@example.com` domain you expect, else send them to a "Your cohort is being onboarded soon / you have no access" page if you have one.

This last approach is feasible (it's how invite-only / referral systems are built for SaaS applications) but is arguably less secure because you have let the world down to the application-level rather than validated access at the Azure service level.

**Comment by olafwrieden** [comment]
Given this thread is only and the initial question has been answered, I will now close the issue 🙂
